atomic_reactor/plugins/post_compare_components.py

```diff
@@ -46,32 +46,18 @@ def rpm_compare(self, a, b):
 
         raise ValueError("%s != %s" % (a, b))
 
-    def get_component_list_from_workers(self, worker_metadatas):
+    def get_rpm_components_list(self):
         """
-        Find the component lists from each worker build.
-
-        The components that are interesting are under the 'output' key.  The
-        buildhost's components are ignored.
-
-        Inside the 'output' key are various 'instances'.  The only 'instance'
-        with a 'component list' is the 'docker-image' instance.  The 'log'
-        instances are ignored for now.
-
-        Reference plugin post_koji_upload for details on how this is created.
+        Get the rpm components list for each platform build and
+        merge it in one list for comparison.
 
         :return: list of component lists
         """
         comp_list = []
-        for platform in sorted(worker_metadatas.keys()):
-            for instance in worker_metadatas[platform]['output']:
-                if instance['type'] == 'docker-image':
-                    if 'components' not in instance or not instance['components']:
-                        self.log.warning(
-                            "Missing 'components' key in 'output' metadata instance: %s", instance
-                        )
-                        continue
+        components_per_platform = self.workflow.data.postbuild_results.get(PostBuildRPMqaPlugin.key)
 
-                    comp_list.append(instance['components'])
+        for components in components_per_platform.values():
+            comp_list.append(components)
```

Comment:
Nitpick: using `postbuild_results.get(key)` suggests that components_per_platform could be None, so calling `components_per_platform.values()` could fail. Could we use `postbuild_results[key]`  or `postbuild_results.get(key, {})` instead?

Reply:
updated to `postbuild_results[key]`